airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -419,91 +319,46 @@ def create_external_table(self,
                              "Please use one of the following types: {1}"
                              .format(compression, allowed_compressions))
 
-        table_resource = {
-            'externalDataConfiguration': {
-                'autodetect': autodetect,
-                'sourceFormat': source_format,
-                'sourceUris': source_uris,
-                'compression': compression,
-                'ignoreUnknownValues': ignore_unknown_values
-            },
-            'tableReference': {
-                'projectId': project_id,
-                'datasetId': dataset_id,
-                'tableId': external_table_id,
-            }
-        }
+        external_config = bigquery.ExternalConfig(source_format)
+        external_config.autodetect = autodetect
+        external_config.compression = compression
+        external_config.source_uris = source_uris
+        external_config.ignore_unknown_values = ignore_unknown_values
 
-        if schema_fields:
-            table_resource['externalDataConfiguration'].update({
-                'schema': {
-                    'fields': schema_fields
-                }
-            })
+        if external_config_options is not None:
+            if not isinstance(external_config_options, type(external_config.options)):
```

Comment:
I think you should use dict explicitly here. What if external_config.options are None (seem to be default).